### PR TITLE
Added ROUNDDOWN formula snippets into the custom defined function.

### PR DESCRIPTION
**Pull request by Vasanth-3519**
**Description:**
Modified the existing user defined function content and added round down formula snippets into the source.

### CODE REVIEW

**Review by gobalakrishnans — CHANGES_REQUESTED** [review]

File: ej2-javascript/code-snippet/spreadsheet/formula-cs1/index.js [hunk 1]
```diff
@@ -38,20 +38,67 @@ function calculatePercentage(firstCell, secondCell) {
...
+      return formulaResult;
+    }
+    // Triggers when the provided digit value is zero.
+    if (valueMode === 0) {
+      formulaResult = splitValue[0];
+      return formulaResult;
+    }
+    // Triggers when the provided digit value is negative.
+    if (valueMode === -1) {
+      //  store zero's upto provided negative digit.
```
[the remark this answers is not in the document]
Comment (Vasanth-3519) [reply ej2-javascript/code-snippet/spreadsheet/formula-cs1/index.js:62]: modified with proper casing

File: ej2-javascript/code-snippet/spreadsheet/formula-cs1/index.ts [hunk 2]
```diff
@@ -44,21 +44,68 @@ function calculatePercentage(firstCell: string, secondCell: string): number {
     return Number(firstCell) / Number(secondCell);
 }
 
+// Custom function to calculate round down for values.
+function RoundDownFunction(value: string, digit: any ) {
```
[the remark this answers is not in the document]
Comment (Vasanth-3519) [reply ej2-javascript/code-snippet/spreadsheet/formula-cs1/index.ts:48]: as the digit may contain positive negative as well as decimal considered as any rtype. Now, modified it.

File: ej2-javascript/code-snippet/spreadsheet/formula-cs1/index.ts [hunk 3]
```diff
@@ -44,21 +44,68 @@ function calculatePercentage(firstCell: string, secondCell: string): number {
...
     sheets: sheets,
     created: (): void => {
         spreadsheet.cellFormat({ fontWeight: 'bold', textAlign: 'center' }, 'A2:E2');
         spreadsheet.numberFormat('$#,##0', 'B3:D12');
         spreadsheet.numberFormat('0%', 'E3:E12');
-
         // Adding custom function for calculating the percentage between two cells.
         spreadsheet.addCustomFunction(calculatePercentage, 'PERCENTAGE');
+        // Adding custom function for calculating round down for the value.
+        spreadsheet.addCustomFunction(RoundDownFunction, 'ROUNDDOWN');
```
[the remark this answers is not in the document]
Comment (Vasanth-3519) [reply ej2-javascript/code-snippet/spreadsheet/formula-cs1/index.ts:98]: changed

File: ej2-javascript/spreadsheet/formulas.md [hunk 4]
```diff
@@ -22,9 +22,13 @@ You can set formula for a cell in the following ways,
 * You can set formula for a cell by [`editing`](./editing).
 * Using the [`updateCell`](../api/spreadsheet/#updatecell) method, you can set or update the cell formula.
 
-## User Defined Functions
+## Create User Defined / Custom Functions
 
-The list of formulas supported in the spreadsheet is sufficient for most of your calculations. If not, you can add your own custom function using the [`addCustomFunction`](../api/spreadsheet/#addcustomfunction) method. Use [`computeExpression`](../api/spreadsheet/#computeexpression) method, if you want to compute any formula or expression.
+Almost all spreadsheet computations can be implemented using the available [`list of supported formulas`](https://ej2.syncfusion.com/documentation/spreadsheet/formulas#supported-formulas).
```
[the remark this answers is not in the document]
Comment (Vasanth-3519) [reply ej2-javascript/spreadsheet/formulas.md:27]: updated

File: ej2-javascript/spreadsheet/formulas.md [hunk 5]
```diff
@@ -22,9 +22,13 @@ You can set formula for a cell in the following ways,
...
+
+If you need formulas that are not supported by spreadsheet, you can add your own custom function using the [`addCustomFunction`](../api/spreadsheet/#addcustomfunction) method. You can also use [`computeExpression`](../api/spreadsheet/#computeexpression) method, if you want to compute any formula or expression.
```
[the remark this answers is not in the document]
Comment (Vasanth-3519) [reply ej2-javascript/spreadsheet/formulas.md:29]: changed

File: ej2-javascript/spreadsheet/formulas.md [hunk 6]
```diff
@@ -22,9 +22,13 @@ You can set formula for a cell in the following ways,
...
+
+This custom function method can be used only for formulas whose results are updated to a single cell value. The formulas that return the result as an array will be tedious to achieve in this case.
```
[the remark this answers is not in the document]
Comment (Vasanth-3519) [reply ej2-javascript/spreadsheet/formulas.md:31]: updated

File: ej2-javascript/code-snippet/spreadsheet/formula-cs1/index.ts [hunk 7]
```diff
@@ -44,21 +44,68 @@ function calculatePercentage(firstCell: string, secondCell: string): number {
...
 
+// Custom function to calculate round down for values.
+function RoundDownFunction(value: string, digit: any ) {
+    let formulaResult: string, sliceValue: string;
+    // Split the decimal values.
+    let splitValue: object = value.split('.');
+    // Fetch the sign of the decimal value.
+    let valueMode: number = Math.sign(parseInt(digit));
+    // Triggers when the provided digit value is positive.
+    if (valueMode === 1) {
```
[the remark this answers is not in the document]
Comment (Vasanth-3519) [reply ej2-javascript/code-snippet/spreadsheet/formula-cs1/index.ts:55]: shared a modified sample for review. If its fine will update it and commit.

**Review by gobalakrishnans — APPROVED** [review]

**Pull request merged**